examples/memspace_numa/memspace_numa.c

```diff
@@ -40,52 +40,89 @@ int createMemoryProvider(umf_memory_provider_handle_t *hProvider,
     if (result != UMF_RESULT_SUCCESS) {
         ret = -1;
         fprintf(stderr, "umfMempolicyCreate failed.\n");
-        goto error_mempolicy;
+        goto error_memspace;
     }
 
     // Create a memory provider using the memory space and memory policy
     result = umfMemoryProviderCreateFromMemspace(hMemspace, hPolicy, hProvider);
     if (result != UMF_RESULT_SUCCESS) {
         ret = -1;
         fprintf(stderr, "umfMemoryProviderCreateFromMemspace failed.\n");
-        goto error_provider;
+        goto error_mempolicy;
     }
 
     // After creating the memory provider, we can destroy the memspace and mempolicy
-error_provider:
-    umfMempolicyDestroy(hPolicy);
 error_mempolicy:
-    umfMemspaceDestroy(hMemspace);
+    umfMempolicyDestroy(hPolicy);
 error_memspace:
+    umfMemspaceDestroy(hMemspace);
     return ret;
 }
 
-int main(void) {
-    umf_memory_provider_handle_t hProvider = NULL;
-    umf_result_t ret;
+// Function to create a memory provider which allocates memory from the specified NUMA node
+// By using filter functions.
```

Comment:
`functions` (plural)? you could be more descriptive, this function only uses the filterById and precisely filter it by 1 numa node;

perhaps you could mention other filter functions available in umf (a link to docs)?

Reply:
i removed plural